### ISSUE TITLE
Databricks: missing support for infix collate notation

### ISSUE THREAD

**Issue by diorge** [sqlfluff/sqlfluff#7941]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The following SQL file fails to parse when using the Databricks dialect:
```sql
CREATE OR REFRESH MATERIALIZED VIEW my_view AS
SELECT
    id COLLATE UTF8_LCASE
FROM my_source;
```

See similar issues:
- #7747 talks about Databricks collate expressions in create table statements, and is fixed in #7749 
- #6917 adds the same support but for Snowflake dialect, fixed in #6967

### Expected Behaviour

The code is perfectly valid Databricks SQL and should parse

### Observed Behaviour

sqlfluff's output:
```sh
== [example.sql] FAIL                                                                                                                                                                                               
L:   2 | P:   1 | LT09 | Select targets should be on a new line unless there is
                       | only one select target. [layout.select_targets]
L:   3 | P:   8 | AL02 | Implicit/explicit aliasing of columns.
                       | [aliasing.column]
L:   3 | P:   8 | CP02 | Unquoted identifiers must be consistently lower case.
                       | [capitalisation.identifiers]
L:   3 | P:   8 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
L:   3 | P:  15 | LT02 | Expected line break and no indent before 'UTF8_LCASE'.
                       | [layout.indent]
L:   3 | P:  16 |  PRS | Line 3, Position 16: Found unparsable section:
                       | 'UTF8_LCASE'
WARNING: Parsing errors found and dialect is set to 'databricks'. Have you configured your dialect correctly?
```

### How to reproduce

Create a example.sql file with the provided sql above and run `sqlfluff lint example.sql --dialect databricks`

### Dialect

Databricks

### Version

Python 3.14.0, sqlfluff 4.2.2

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
